### PR TITLE
transfer previous tests to  v0.0.1-misc

### PR DESCRIPTION
**Pull request by MrLYG**

### CODE REVIEW

File: tests/utils/test_file.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+def test_nested_set_path_not_exist():
+    dict_ = {'a': 1}
+    keys = ['b', 'c']
+    value = 2
+    expected_output = {'a': 1, 'b': {'c': 2}}
+    # Test `nested_set` with keys not in dict to check if it add the keys and value to the dict
+    assert nested_set(dict_, keys, value) == expected_output
+
+
+def test_truncate_utf8_chars_not_found():
```
Comment (fostiropoulos) [inline tests/utils/test_file.py]: please use tmp_path

File: tests/modules/test_summary_logger.py [hunk 2]
```diff
@@ -260,9 +280,22 @@ def test_results_json(tmp_path: Path):
...
+def test_summary_logger_with_none():
+    l = SummaryLogger(c)
+
+    # Test _make_dashboard function with `summary_dir` is None.
+    # To check if it can cover edge case.
+    assert l._make_dashboard(summary_dir=None) is None
+
+    # Test _write_config function with `model_dir` is None
+    # To check if it can cover edge case.
+    assert l._write_config(c) is None
```
Comment (fostiropoulos) [inline tests/modules/test_summary_logger.py]: this is not a valid test case

File: tests/modules/test_summary_logger.py [hunk 3]
```diff
@@ -260,9 +280,22 @@ def test_results_json(tmp_path: Path):
     # return
     pass
 
+def test_summary_logger_with_none():
+    l = SummaryLogger(c)
+
+    # Test _make_dashboard function with `summary_dir` is None.
+    # To check if it can cover edge case.
+    assert l._make_dashboard(summary_dir=None) is None
```
Comment (fostiropoulos) [inline tests/modules/test_summary_logger.py]: this is not a valid test case

File: tests/modules/test_optim_scheduler.py [hunk 4]
```diff
@@ -56,6 +58,67 @@ def test_optimizers():
...
+    assert len(list(model.parameters())) == len(params)
+    for p1, p2 in zip(model.parameters(), params):
+        assert torch.all(p1 == p2)
+
+
+# This will test whether the function correctly extracts all the parameters from the model
+# and whether these are the correct parameters.
+# This test case is written for the scenario where `only_requires_grad`` is False.
+def test_get_optim_parameters_without_decay_and_with_all_parameters():
+    model = torch.nn.Linear(10, 1)
```
Comment (fostiropoulos) [inline tests/modules/test_optim_scheduler.py]: can we use a more complex model (you can look at `ray_models` module)?

**Review by fostiropoulos — CHANGES_REQUESTED** [review]

**Comment by fostiropoulos** [comment]
@Rudra-23 can you please review.

File: tests/utils/test_base.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+    for error in non_oom_errors:
+        assert base.is_oom_exception(RuntimeError(error)) is False, f"Expected False for error message '{error}'"
+
+
+# Test If the apply_lambda_to_iter function correctly applies a lambda function
+# (in this case, a function to square a number) to each element in a list
+def test_apply_lambda_to_iter():
+    input_list = [1, 2, 3, 4, 5]
+    expected_output = [1, 4, 9, 16, 25]
+    output = base.apply_lambda_to_iter(input_list, lambda x: x**2)
```
Comment (Rudra-23) [inline tests/utils/test_base.py:113]: The function ``apply_lambda_to_iter`` takes an Iterable: which is list, dict, str and tuple. The list part is checked. However, tests on other types (especially ``dict``) can be be added.
